crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -998,7 +985,7 @@ impl std::fmt::Display for DiagnosticId {
 ///
 /// This enum presents a unified interface to these two types for the sake of creating [`Span`]s and
 /// emitting diagnostics from both ty and ruff.
-#[derive(Debug, Clone, PartialEq, Eq, get_size2::GetSize)]
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, get_size2::GetSize)]
```

Comment:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, get_size2::GetSize)]
```

Yeah, I think I've changed my opinion after looking through the snapshots. The new sort order is sort of confusing (Unless we make `RenderingSortKey` parametrizable on whether it should use the `secondary_code` first before using the `id`).